Unreachable if-branches found with nilness analyzer

I tried running the [nilness](https://pkg.go.dev/golang.org/x/tools/go/analysis/passes/nilness#cond) analyzer using `go vet -vettool $(which nilness) ./...` which resulted in these matches. When looking at the code, it does seem a bit peculiar.

```
# helm.sh/helm/v4/pkg/downloader
# [helm.sh/helm/v4/pkg/downloader]
pkg/downloader/chart_downloader.go:234:9: impossible condition: nil != nil
# helm.sh/helm/v4/pkg/repo/v1
# [helm.sh/helm/v4/pkg/repo/v1]
pkg/repo/v1/index_test.go:711:12: impossible condition: nil != nil
```

This `pkg/repo/v1/index_test.go:711` 
```go
			idx, err := loadIndex([]byte(tc.data), tc.source)
			if err != nil {
				t.Fatalf("unexpected error: %s", err)
			}
			cvs := idx.Entries["nginx"]
			if cvs == nil {
                               # err is never re-assigned, so it will remain nil.
				if err != nil {
					t.Error("expected one chart version not to be filtered out")
				}
			}
```

For `pkg/downloader/chart_downloader.go:234`, there is a similar logic. `err` is never re-assigned, so it's dead/unreachable code:

```go
	if err != nil {
		return "", nil, err
	}
	var digest32 [32]byte
	copy(digest32[:], digest)
	if err != nil {
		return "", nil, fmt.Errorf("unable to decode digest: %w", err)
	}
```

Perhaps something is missing before the if-check in the two code snippets.

--- a/pkg/downloader/chart_downloader.go
+++ b/pkg/downloader/chart_downloader.go
@@ -227,13 +227,10 @@ func (c *ChartDownloader) DownloadToCache(ref, version string) (string, *provena
 	// Check the cache for the file
 	digest, err := hex.DecodeString(digestString)
 	if err != nil {
-		return "", nil, err
+		return "", nil, fmt.Errorf("unable to decode digest: %w", err)
 	}
 	var digest32 [32]byte
 	copy(digest32[:], digest)
-	if err != nil {
-		return "", nil, fmt.Errorf("unable to decode digest: %w", err)
-	}
 
 	var pth string
 	// only fetch from the cache if we have a digest

--- a/pkg/repo/v1/index_test.go
+++ b/pkg/repo/v1/index_test.go
@@ -708,9 +708,7 @@ func TestLoadIndex_DuplicateChartDeps(t *testing.T) {
 			}
 			cvs := idx.Entries["nginx"]
 			if cvs == nil {
-				if err != nil {
-					t.Error("expected one chart version not to be filtered out")
-				}
+				t.Error("expected one chart version not to be filtered out")
 			}
 			for _, v := range cvs {
 				if v.Name == "alpine" {

